code/modules/mob/living/carbon/human/human_defense.dm

```diff
@@ -172,6 +172,13 @@
 	else
 		affecting = get_bodypart(ran_zone(user.zone_selected))
 	var/target_area = parse_zone(check_zone(user.zone_selected)) //our intended target
+	if(affecting)
+		if(I.force && I.damtype != STAMINA && affecting.status == BODYPART_ROBOTIC) //Bodypart_Robotic sparks when hit, but only when it does real damage.

```

Comment:
register a component listener and just listen for COMSIG_ITEM_ATTACK_ZONE on the IPC

I mean jesus christ the signal is literally right below this code, there's no excuse for this